syft/frameworks/torch/tensors/interpreters/chinese_remainder_theorem.py

```diff
@@ -0,0 +1,394 @@
+import torch
+import math
+import itertools
+
+import syft
+from syft.frameworks.torch.tensors.interpreters.abstract import AbstractTensor
+from syft.frameworks.torch.tensors.interpreters.precision import FixedPrecisionTensor
+from syft.frameworks.torch.tensors.interpreters.additive_shared import AdditiveSharingTensor
+from syft.frameworks.torch.overload_torch import overloaded
+
+
+class CRTTensor(AbstractTensor):
+    """ A CRT tensor is a tensor whose values are represented as their remainders modulo several pairwise coprime numbers.
+    The true tensor values lie in the Zq field where q is the product of all the moduli.
+    A CRT tensor then represent a modular system of equations:
+        x = a_0 mod f_0
+        ...
+        x = a_n mod f_n
+    The Chineses Remainder Theorem (that also gives his name to this tensor) assert that exactly one x in Zq with q = f_0 * ... * f_n
+    satisfies this system. This x is the real value represented by the tensor.
+    This tensor makes it possible to represent big numbers and to avoid overflows while manipulating them.
+    It also makes additions, subtractions, and multiplications of huge numbers quite efficient.
+    """
+
+    def __init__(
+        self,
+        field: int = 21,
+        residues: dict = None,
+        base=None,
+        precision_fractional=None,
+        reconstruction_coeffs=None,
+        owner=None,
+        id=None,
+        tags: set = None,
+        description: str = None,
+    ):
+        super().__init__(owner=owner, id=id, tags=tags, description=description)
+
+        # Check that all the moduli are pairwise coprime
+        if residues is not None:
+            for pair in itertools.combinations(residues.keys(), r=2):
+                assert (
+                    math.gcd(pair[0], pair[1]) == 1
+                ), f"{pair[0]} and {pair[1]} are not coprime, you cannot build a CRTTensor with these as moduli"
+
+        self.field = field
+
+        # Check that all the residues have the same precision and
+        # check that all the shapes are the same
+        # while filling tensor
+        self.base = base
+        self.precision_fractional = precision_fractional
+        if residues is not None:
+            r = next(iter(residues.values()))  # Take one arbitrary residue
+            assert isinstance(r.child, FixedPrecisionTensor)
```

Comment:
Can you add a human readable message for this assertion so that people know what needs to be passsed